Source/LinqToDB.EntityFrameworkCore/LinqToDBForEFTools.ContextExtensions.cs

```diff
@@ -257,7 +261,51 @@ public static ITable<T> GetTable<T>(this DbContext context)
 
 			return context.CreateLinqToDbContext().GetTable<T>();
 		}
-		
+
+		#endregion
+
+		#region Interceptors
+
+		/// <summary>
+		/// Returns list of registered Linq2Db interceptors from EF Context
+		/// </summary>
+		/// <typeparam name="T">Mapping class type.</typeparam>
+		/// <returns>Db context object</returns>
+		public static IList<IInterceptor> GetLinq2DbInterceptors(this DbContext context)
+		{
+			if (context == null) throw new ArgumentNullException(nameof(context));
+
+			var contextOptions = ((IInfrastructure<IServiceProvider>)context.Database)?
+				.Instance?.GetService(typeof(IDbContextOptions)) as IDbContextOptions;
+
+			return contextOptions?.GetLinq2DbInterceptors() ?? new List<IInterceptor>();
+		}
+
+		/// <summary>
+		/// Returns list of registered Linq2Db interceptors from EF Context options
+		/// </summary>
+		/// <typeparam name="T">Mapping class type.</typeparam>
+		/// <returns>Db context options</returns>
+		public static IList<IInterceptor> GetLinq2DbInterceptors(this IDbContextOptions contextOptions)
+		{
+			if (contextOptions == null) throw new ArgumentNullException(nameof(contextOptions));
+
+			var linq2DbExtension = contextOptions?.FindExtension<LinqToDBOptionsExtension>();
+			var coreEfExtension = contextOptions?.FindExtension<CoreOptionsExtension>();

```

Comment:
What you can do with EF Interceptors?

Reply:
Look at the tests - you can for example define EF Interceptors which implements both EF interface and Linq2Db interface. That way you have one interceptor which will be taken into consideration in both EF logic and Linq2Db logic.

Reply:
So, then maybe better to add them twice? It is hidden behavior.

Reply:
You can add them twice, but I made this to further simplify the design. When you have a working EF code with interceptors all you need to do is to just implement the interface from Linq2Db. In my opinion this would be quite an improvement.

Reply:
Instead of enumerating one array we start doing something unusual. And I'm almost sure EF users will not mix linq2db code with EF.

Reply:
In this I would disagree, ability to have one interceptor class that for example adds to every command a line of code it was executed from instead of two is in my opinion great and we use that in our projects. Adding this functionality to be able to consume them automatically without any additional configuration is a benefit in my opinion.

Reply:
What I don't like that removing that in the future will be breaking change ;) linq2db registers own interceptors, EF Core register it's own interceptors. Not strictly disagree about this feature but trying to avoid side effects and supporting efforts.

Reply:
So basically, you want to always set the linq2db interceptors explicitly in the context options to prevent misuse? I have a suggestion - what about having this ablity coded but disabled by default? Then I would add a separate setting in opsions to enable it, so for example it would look that way:
```
optionsBuilder.UseLinqToDb(builder => 
			{
				builder.TryToUseEfCoreRegisteredInterceptors()
			});
```

Reply:
Much better

Reply:
Done - please take a look.

Reply:
Well, I do not agree that this is a lot of extra logic - basically 3 lines of code. Like I said above with conversation with @sdanyliv  - there is a real use case for this and the new method (along with its description) in my opinion clarifies what it does and how it works.

Reply:
I changed the name of the method to **UseEfCoreRegisteredInterceptorsIfPossible** - I think it better describes what it actually does.